staging/src/k8s.io/apiserver/pkg/endpoints/handlers/response.go

```diff
@@ -141,28 +143,30 @@ func (e *watchEmbeddedEncoder) embeddedIdentifier() runtime.Identifier {
 //
 // NOTE: watchEncoder is NOT thread-safe.
 type watchEncoder struct {
-	ctx             context.Context
-	kind            schema.GroupVersionKind
-	embeddedEncoder runtime.Encoder
-	encoder         runtime.Encoder
-	framer          io.Writer
+	ctx                    context.Context
+	kind                   schema.GroupVersionKind
+	watchObjectsEncoder    runtime.Encoder
+	watchEventsEncoder     runtime.Encoder
+	framer                 io.Writer
+	watchListTransformerFn watchListTransformerFunction
```

Comment:
Please don't over-generalize.
There is exactly one transformation that we're doing here, so let's not add additional fields, just have it hardcoded in the place where this is called.

Reply:
Does it mean you would like to pass `initialEventsListBlueprint runtime.Object` instead ?